tests/integration/se-gh-6845/src/main/java/io/helidon/tests/integration/gh6845/Main.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.tests.integration.gh6845;
+
+import io.helidon.common.LogConfig;
+import io.helidon.common.http.DataChunk;
+import io.helidon.common.reactive.Single;
+import io.helidon.media.multipart.ContentDisposition;
+import io.helidon.media.multipart.MultiPartSupport;
+import io.helidon.media.multipart.WriteableBodyPart;
+import io.helidon.media.multipart.WriteableBodyPartHeaders;
+import io.helidon.media.multipart.WriteableMultiPart;
+import io.helidon.webserver.Routing;
+import io.helidon.webserver.WebServer;
+
+import java.util.Arrays;
+
+import static io.helidon.common.http.MediaType.TEXT_PLAIN;
+
+/**
+ * Main class of this integration test.
+ */
+public final class Main {
+
+    /**
+     * Cannot be instantiated.
+     */
+    private Main() {
+    }
+
+    /**
+     * Application main entry point.
+     *
+     * @param args command line arguments.
+     */
+    public static void main(String[] args) {
+        startServer();
+    }
+
+    /**
+     * Start the server.
+     *
+     * @return the created {@link io.helidon.webserver.WebServer} instance
+     */
+    static WebServer startServer() {
+        // load logging configuration
+        LogConfig.configureRuntime();
+
+        WebServer server = WebServer.builder()
+                                    .routing(createRouting())
+                                    .addMediaSupport(MultiPartSupport.create())
+                                    .printFeatureDetails(true)
+                                    .build();
+
+        // Try to start the server. If successful, print some info and arrange to
+        // print a message at shutdown. If unsuccessful, print the exception.
+        server.start()
+              .thenAccept(ws -> {
+                  System.out.println(
+                          "WEB server is up! http://localhost:" + ws.port() + "/greet");
+                  ws.whenShutdown().thenRun(()
+                          -> System.out.println("WEB server is DOWN. Good bye!"));
+              })
+              .exceptionally(t -> {
+                  System.err.println("Startup failed: " + t.getMessage());
+                  t.printStackTrace(System.err);
+                  return null;
+              });
+
+        // Server threads are not daemon. No need to block. Just react.
+        return server;
+    }
+
+    /**
+     * Creates new {@link io.helidon.webserver.Routing}.
+     *
+     * @return routing configured
+     */
+    private static Routing createRouting() {
+        return Routing.builder()
+                      .get("/", (req, res) -> {
+                          byte[] contents = new byte[21 << 20 - 1];
```

Comment:
Comment explaining the magic number for the future could be appreciated some day :smile:

Reply:
`21 << 20` is 21 MiB in bytes.

```java
1 << 10 // 2 ^ 10 = 1024
1 << 20 // 2 ^ 20 = 1024 * 1024
21 << 20 // 21 * 2 ^ 20 = 21 * 1024 * 1024
```

TBH I took it as-is from the person that reported the initial issue. I'm not sure what the `-1` is for, maybe this was meant for less than `21 MiB` but in that case it should have been `(21 << 20) - 1`